src/zimscraperlib/rewriting/html.py

```diff
@@ -132,9 +132,9 @@ class HtmlRewriter(HTMLParser):
     def __init__(
         self,
         url_rewriter: ArticleUrlRewriter,
-        pre_head_insert: str,
+        pre_head_insert: str | None,
         post_head_insert: str | None,
-        notify_js_module: Callable[[ZimPath], None],
+        notify_js_module: Callable[[ZimPath], None] | None,
```

Comment:
Out of curiosity, wouldn't it be more maintainable to pass a no-op callable in those cases? No need for conditionals everywhere and makes the type check happier

Reply:
This is what I did before (implement a no-op), and I thought it was indeed better. Now that I'm using this in mindtouch scraper where we do not care about JS, it is really weird to have to implement / pass a no-op, so I decided it would be wise to support "no notify_js_module", also because it is clearer in mindtouch scraper. 
Clearly not a very strong argument, and obviously a trade-off.